sharktank/sharktank/layers/configs/llm_configs.py

```diff
@@ -88,18 +172,33 @@ def from_gguf_props(p: dict[str, Any]):
             attention_head_count_kv=_optional_int_prop(
                 p, f"{name_prefix}.attention.head_count_kv", attention_head_count
             ),
-            attn_head_dim=rope_dimension_count,
+            attn_head_dim=attn_head_dim,
```

Comment:
For any of these new parameters that are deepseek specific we can do the following.

If we have a dict with `{"q_lora_rank":q_lora_rank}` we can just pass this to the constructor `**architecture_specialization`. This expansion just lets architecture specific customization to be passed around as dictionaries instead.